recipes-openxt/openxt/openxt-keymanagement/key-functions

```diff
@@ -0,0 +1,551 @@
+        return 0;
+    else
+        return 1;
+    fi
+}
+
+close_keystore() {
+    if grep -q ${KEYMNT} /proc/mounts; then
+        umount ${KEYMNT}
+    fi
+}
+
+password_prompt() {
+    local request=$1
+    local key=$2
+
+    exec 3>/dev/tty
+
+    if ! keystore_ready; then
+        dialog --ok-label Continue --msgbox \
+            "ERROR: Keystore does not appear to be available!" 0 0 1>&3
+        return 1;
+    fi
+
+    while [ 1 ] ; do
+        local capture="${KEYDIR}/${key}"
+        dialog --cancel-label Previous --insecure --passwordbox \
+            "Please enter a new password for ${request}" 0 50 "" 2>${capture} 1>&3
+
+        local ret=$?
+        if [ ${ret} != 0 ]; then
+            rm -f ${capture}
+            exec 3>&-
+            return ${ret}
+        fi
+
+        local repeat=$(mktemp -p ${KEYDIR})
+        dialog --cancel-label Previous --insecure --passwordbox \
+            "Please enter the same password to confirm" 0 50 "" 2>${repeat} 1>&3
+
+        local ret=$?
+        if [ ${ret} != 0 ]; then
+            rm -f ${capture}
+            rm -f ${repeat}
+            exec 3>&-
+            return ${ret}
+        fi
+
+        local password=$(cat ${capture})
+
+        if [ -z "${password}" ]; then
+            rm -f ${capture}
+            rm -f ${repeat}
+            dialog --colors --ok-label Continue --msgbox \
+                '  \ZbERROR:\ZB Empty password!\n\n    Please try again.' 7 30 1>&3
+            continue
+        fi
+
+        if diff -q ${capture} ${repeat} 2>/dev/null ; then
+            rm -f ${repeat}
+            break
+        fi
+        rm -f ${repeat}
+
+        dialog --ok-label Continue --msgbox \
+            "ERROR: Passwords did not match. Please try again." 0 0 1>&3
+    done
+
+    exec 3>&-
+    echo ${capture}
+}
+
+recovery_prompt() {
+    local try=""
+
+    # Explicit interact on tty1
+    exec 3>/dev/tty1
+
+    if ! keystore_ready; then
+        dialog --ok-label Continue --msgbox \
+            "ERROR: Keystore does not appear to be available!" 0 0 1>&3
+        return 2;
+    fi
+
+    case ${1} in
+        1) try="first" ;;
+        2) try="second" ;;
+        3) try="third" ;;
+        *) try="more than you should have" ;;
+    esac
+
+    local capture="${KEYDIR}/recovery"
+    dialog --no-cancel --insecure --colors --passwordbox "
+    This is the ${try} attempt.
+
+    Please enter the recovery password " 0 0 2>${capture} 1>&3
+
+    local ret=$?
+    if [ ${ret} != 0 ]; then
+        rm -f ${capture}
+        exec 3>&-
+        return ${ret}
+    fi
+
+    exec 3>&-
+    echo ${capture}
+}
+
+# Seed kernel's CPRNG with some entropy
+# This is a cheap attempt to get the entropy pool to an acceptable
+#   level. If there is a concern then override this function to call
+#   out to an external entropy source.
+seed_entropy() {
+    local seeks="5 72 96 128 51 83 112 12 31 205"
+
+    if [[ ! -e /dev/sda ]]; then
```

Comment:
I don't have an opinion, I just think code should be consistent, especially in new files.
`[[` might be a tiny bit faster, since it's a builtin and not an external binary IIRC.